ethexe/signer/src/lib.rs

```diff
@@ -170,59 +159,62 @@ impl fmt::Display for PublicKey {
     }
 }
 
-impl fmt::Display for Address {
-    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        write!(f, "0x{}", self.to_hex())
-    }
-}
-
+/// Signer which signs data using owned key store.
 #[derive(Debug, Clone)]
 pub struct Signer {
     key_store: PathBuf,
 }
 
 impl Signer {
+    /// Create a new signer with a key store location.
     pub fn new(key_store: PathBuf) -> Result<Self> {
         fs::create_dir_all(key_store.as_path())?;
 
         Ok(Self { key_store })
     }
 
+    /// Create a new signer with a key temporary key store location.
     pub fn tmp() -> Self {
         let temp_dir = tempfile::tempdir().expect("Cannot create temp dir for keys");
         Self {
             key_store: temp_dir.into_path(),
         }
     }
 
+    /// Create a ECDSA recoverable signature with `Electrum` notation for the `v` value.
+    ///
+    /// For more info about `v` value read [`RawSignature`] docs.
     pub fn raw_sign_digest(&self, public_key: PublicKey, digest: Digest) -> Result<RawSignature> {
         let private_key = self.get_private_key(public_key)?;
 
         RawSignature::create_for_digest(private_key, digest)
     }
 
+    /// Create a ECDSA recoverable signature.
+    // TODO #4365
     pub fn sign_digest(&self, public_key: PublicKey, digest: Digest) -> Result<Signature> {
         let private_key = self.get_private_key(public_key)?;
 
         Signature::create_for_digest(private_key, digest)
     }
 
+    /// Create a ECDSA recoverable signature for the raw bytes data.
     pub fn sign(&self, public_key: PublicKey, data: &[u8]) -> Result<Signature> {
         self.sign_digest(public_key, data.to_digest())
     }
 
+    /// Create a ECDSA recoverable signature for the raw bytes data with
+    /// an ethereum address provided instead of the public key.
+    ///
+    /// If the private key for the ethereum address is stored, the signature will be returned.
     pub fn sign_with_addr(&self, address: Address, data: &[u8]) -> Result<Signature> {
-        let keys = self.list_keys()?;
-
-        for key in keys {
-            if key.to_address() == address {
-                return self.sign(key, data);
-            }
+        match self.get_key_by_addr(address)? {
+            Some(public_key) => self.sign(public_key, data),
+            None => bail!("Address not found: {}", address),
```

Comment:
```suggestion
            None => anyhow!("Address not found: {address}"),
```

Reply:
it should return `Err(anyhow!(...))`, which is the same as `bail!`